syntax = "proto3";

import "person_info.proto";

package persons;

message Person {
    PersonInfo info = 1; // characteristics of the person
    // The "repeated" keyword indicates a variable that is interpreted as a vector (Array/List). In this case, "friends" is a vector of "Friend" messages.
    repeated Friend friends = 2; // friends of the person
}

message Friend {
    float friendship_duration = 1; // duration of friendship in days
    repeated string shared_hobbies = 2; // shared interests
    Person person = 3; // identity of the friend
}